File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        i,j = 0, len(matrix[0])-1
        while i<len(matrix) and j>=0 and matrix[i][j]!=target:
            if target>matrix[i][j]:
                i+=1
            else:
                j-=1
        return False if (i>=len(matrix) or j<0) else True
        

File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int i=0, j=matrix[0].length-1;
        while(i<matrix.length && j>=0 && target!=matrix[i][j]){
            if(target>matrix[i][j]) i++;
            else j--;
        }
        return (i>=matrix.length || j<0) ? false : true;
    }
}